### PR TITLE
Fix the CallApplicationMain's return type (#850)

### PR DESCRIPTION
**Pull request by troydai**
Addressing issue #850 

`klr` except following signature of `CallApplicationMain`
https://github.com/aspnet/KRuntime/blob/dev/src/klr/klr.h#L11

### CODE REVIEW

File: src/klr.core45/klr.core45.cpp [hunk 1]
```diff
@@ -498,8 +498,8 @@ extern "C" __declspec(dllexport) bool __stdcall CallApplicationMain(PCALL_APPLIC
     {
         wprintf_s(L"TPA      %d %S\n", wcslen(pwszTrustedPlatformAssemblies), pwszTrustedPlatformAssemblies);
         wprintf_s(L"AppPaths %S\n", wszAppPaths);
-        printf_s("Failed to create app domain (%d).\n", hr);
-        return false;
+        printf_s("Failed to create app domain (%d)!\n", hr);
```
Comment (troydai) [inline src/klr.core45/klr.core45.cpp]: Will revert this line.

**Comment by davidfowl** [comment]
You need to check all of the callers (like Helios) to make sure they don't explode

**Comment by troydai** [comment]
@GrabYourPitchforks

**Comment by troydai** [comment]
Like this https://github.com/aspnet/Helios/blob/21bb86ede13551c1b09074a21e5e66f97ad8c992/src/Microsoft.AspNet.Loader.IIS.Interop/helios.h#L160?

I'll try out the Helios scenarios. Any other callers we know?

**Comment by troydai** [comment]
Helios works fine.

**Comment by davidfowl** [comment]
:shipit:

**Comment by troydai** [comment]
Rebase and squashed. Running last round of verification before merge to dev.

**Pull request merged**